python/feathub/feature_views/feature_view.py

```diff
@@ -136,18 +150,22 @@ def get_output_fields(self, source_fields: List[str]) -> List[str]:
 
         return list(OrderedDict.fromkeys(reordered_output_fields))
 
-    def get_feature(self, feature_name: str) -> Feature:
+    def get_output_features(self) -> List[Feature]:
         if self.is_unresolved():
             raise RuntimeError("Build this feature view before getting features.")
-        for feature in self.get_resolved_features():
-            if feature_name == feature.name:
-                return feature
 
-        if not self.keep_source_fields:
-            raise RuntimeError(
-                f"Failed to find the feature '{feature_name}' in {self.to_json()}."
-            )
-        return cast(TableDescriptor, self.source).get_feature(feature_name)
+        source_features = self.get_resolved_source().get_output_features()
+        features = {
+            **{f.name: f for f in source_features},
+            **{f.name: f for f in self.get_resolved_features()},
+        }
+
+        output_feature_names = self.get_output_fields([f.name for f in source_features])
```

Comment:
Would it be simpler to remove `FeatureView#get_output_fields`?

Only OnDemandFeatureView needs `get_output_fields(self, source_fields: List[str])` because it does not have source information available when it is constructed. We can add `OnDemandFeatureView#get_output_features(self, source_fields: List[str])`.

Please feel free to create a TODO for this task if you prefer.

Reply:
I added a TODO.